mirascope/tools/base.py

```diff
@@ -37,28 +39,75 @@ class ConfigurableTool(
     model_config = ConfigDict(arbitrary_types_allowed=True)
 
     @classmethod
-    def _config(cls) -> _ToolConfigT:
+    def _get_config(cls) -> _ToolConfigT:
```

Comment:
I understand we need this method to get the correct type, but any particular reason to update this to `_get_config`? I generally think of using "get" when there's also a setter, but in this case there should never be a setter (only the `from_config` method).

Also, something I've been thinking about is the fact that `BaseTool` already has `tool_config` as a ClassVar, and I'm starting to think that `__config__` and `tool_config` as ClassVars may become confusing?

WDYT? Any idea here around how to remove confusion?

I think `from_config` and `config` vs. `tool_config` is clear enough, but we type `config: _ToolConfigT` which is where the confusion arises for me (because that has type `ToolConfig`).

maybe we could rename this to something like `_ConfigurableToolConfig`?

Reply:
> I understand we need this method to get the correct type, but any particular reason to update this to _get_config? I generally think of using "get" when there's also a setter, but in this case there should never be a setter (only the from_config method).

By not making the name a verb, we found it difficult to recognize whether it is callable or not.

>　Also, something I've been thinking about is the fact that BaseTool already has tool_config as a ClassVar, and I'm starting to think that __config__ and tool_config as ClassVars may become confusing?

I agree with this. It is indeed difficult to understand.

> maybe we could rename this to something like _ConfigurableToolConfig?

Sounds good, let's give it a try.

Reply:
Ok, `_get_config` is fine then, that makes sense